embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/session/OtelSessionGatingTest.kt

```diff
@@ -0,0 +1,142 @@
+import io.embrace.android.embracesdk.fakes.injection.FakeDeliveryModule
+import io.embrace.android.embracesdk.gating.EmbraceGatingService
+import io.embrace.android.embracesdk.gating.GatingService
+import io.embrace.android.embracesdk.gating.SessionGatingKeys
+import io.embrace.android.embracesdk.getSentSessionMessages
+import io.embrace.android.embracesdk.internal.spans.EmbraceSpanData
+import io.embrace.android.embracesdk.payload.SessionMessage
+import io.embrace.android.embracesdk.recordSession
+import io.embrace.android.embracesdk.session.orchestrator.SessionSnapshotType
+import io.embrace.android.embracesdk.spans.EmbraceSpanEvent
+import io.embrace.android.embracesdk.verifySessionHappened
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * Asserts that OTel parts of sessions are gated.
+ */
+@RunWith(AndroidJUnit4::class)
+internal class OtelSessionGatingTest {
+
+    private var gatingConfig = SessionRemoteConfig(
+        fullSessionEvents = setOf(),
+        sessionComponents = setOf()
+    )
+
+    private val gatingService = EmbraceGatingService(
+        FakeConfigService(
+            sessionBehavior = fakeSessionBehavior {
+                RemoteConfig(sessionConfig = gatingConfig)
+            }
+        )
+    )
+
+    @Rule
+    @JvmField
+    val testRule: IntegrationTestRule = IntegrationTestRule(
+        harnessSupplier = {
+            Harness(
+                fakeDeliveryModule = FakeDeliveryModule(
+                    deliveryService = GatedDeliveryService(gatingService)
+                )
+            )
+        }
+    )
+
+    @Before
+    fun setUp() {
+        assertTrue(testRule.harness.getSentSessionMessages().isEmpty())
+    }
+
+    @Test
+    fun `session sent in full without gating`() {
+        gatingConfig = SessionRemoteConfig()
+
+        with(testRule) {
+            simulateSession()
+            val payload = harness.getSentSessionMessages().single()
+            verifySessionHappened(payload)
+            assertSessionGating(payload, gated = false)
+        }
+    }
+
+    @Test
+    fun `session gated`() {
+        gatingConfig = SessionRemoteConfig(
+            sessionComponents = emptySet(),
+            fullSessionEvents = setOf(
+                SessionGatingKeys.FULL_SESSION_CRASHES,
+                SessionGatingKeys.FULL_SESSION_ERROR_LOGS
+            )
+        )
+
+        with(testRule) {
+            simulateSession()
+            val payload = harness.getSentSessionMessages().single()
+            verifySessionHappened(payload)
+            assertSessionGating(payload, gated = true)
+        }
+    }
+
+    @Suppress("UNUSED_PARAMETER")
+    private fun assertSessionGating(
+        payload: SessionMessage,
+        gated: Boolean
```

Comment:
Why are we passing this if it's not used? Do you plan to use it in the future?

Reply:
Yep - I'm using it in #406. I've broken it down here to make the changeset smaller